assets/src/edit-story/utils/getMediaBaseColor.js

```diff
@@ -61,6 +61,7 @@ function setOrCreateImage(src, onLoad) {
   img.height = 'auto';
   img.onload = onLoad;
   // For 3rd-party media.
+  // todo: setting crossOrigin here causes an error for gif backgrounds
```

Comment:
```suggestion

```

Reply:
I'll leave these suggestions here in case removing it will work fine. It seems to get rid of the error pointed out and doesn't affect the preview.

Reply:
Hmm, actually `img.crossOrigin = 'anonymous';` was originally added for that specific issue -- since otherwise, it gave a cross-origin error. Is this an issue for gifs only? And do the images still seem to work OK without the ` anonymous`?

Reply:
@miina setting a gif background with `img.crossOrigin = 'anonymous'` throws the following error:
``1Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://c.tenor.com/tt5sPyFZh5sAAAAC/%E0%B8%95%E0%B8%9A%E0%B8%A1%E0%B8%B7%E0%B8%AD-%E0%B8%9B%E0%B8%A3%E0%B8%9A%E0%B8%A1%E0%B8%B7%E0%B8%AD.gif. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing).```
(but doesn't affect any rendering in the editor or the preview)
![Screen Shot 2020-10-22 at 10 16 05 AM](https://user-images.githubusercontent.com/41136059/96907106-a5b4e600-144f-11eb-867e-29cef38cb9e8.png)

But also, when I remove the call to set `img.crossOrigin`, I don't see any CORS errors for dropping video, gif, or images into the background (and everything renders in the preview locally).

I'm going to remove it for now, but I'm curious why I'm not seeing the cross-origin error.

Reply:
Interestingly, removing `crossOrigin` breaks the karma tests... trying to figure out why

Reply:
```TypeError: Cannot read property 'parentElement' of null```

In order to fix this, I'm going to change `getMediaBaseColor` to check for the `gif` type and provide a crossOrigin conditionally for now...

Reply:
I removed a `waitFor...` call which seemed to address the issues with the tests. And haven't detected any cross-origin errors.

Reply:
Hmm, okay, odd, but great that it works like this now.

Other thing I noticed is that the 3 party media images are not actually getting a base color assigned either... Doesn't seem to be happening on the `main` branch either though. Is it working for you?
Will create a separate issue to look into that if not.

Reply:
@miina after some investigation, the catch here is firing in `getMediaBaseColor`:
```
export function getMediaBaseColor(resource, onBaseColor) {
  const { type, poster, src } = resource;
  const onLoad = () => {
    const node = document.body[BASE_COLOR_NODE];
    try {
      onBaseColor(thief.getColor(node.firstElementChild));
    } catch (e) {
      onBaseColor([255, 255, 255]);
    }
  };
  setOrCreateImage('video' === type ? poster : src, onLoad);
}```

The error is a `DOMException`, `The operation is insecure.`

Reply:
For the above error, appears that the call `thief.getColor` works fine, but the callback causes the error. in this case it was `finalizeDrop` in `handleDrop`.

Reply:
Hmm, when I'm testing on `main` then it works as expected though currently 😕 The color is received without issues and displays in the FE as expected. Will test the PR again, too ...

Reply:
@embarks I tested again this PR vs `main` in `https://stories-qa-wordpress-amp.pantheonsite.io/` and it seems that on the `main` branch the background color is assigned for 3rd party media, however, on this branch it's not assigned anymore, the `default white background is assigned instead.

Reply:
Ok, I'll have to test again. If white was assigned that catch i pointed out must be the one thats catching, which means the image's onload is throwing an error. 

There's an innerHTML assignment in that function and i think Firefox has a security problem with that. What browser are you using @miina? I'm not really sure what to do about the crossorigin policy now if it's breaking this regression... Maybe add it back for images except gif types? Might have to address the bug for this later.

Reply:
I'm using Chrome, forgot to mention that. 
Tested on FF as well just now on the QA environment (`https://stories-qa-wordpress-amp.pantheonsite.io/`) and same story here -- background color fails for 3party media on the PR but is assigned as expected on `main`.